quic/s2n-quic-core/src/frame/dc_stateless_reset_tokens.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{frame::ExtensionTag, stateless_reset, varint::VarInt};
+use s2n_codec::{decoder_invariant, decoder_parameterized_value, Encoder, EncoderValue};
+
+const TAG: VarInt = VarInt::from_u32(0xdc0000);
+
+macro_rules! dc_stateless_reset_tokens_tag {
+    () => {
+        0xdc0000u64
+    };
+}
+
+//# DC_STATELESS_RESET_TOKENS Frame {
+//#   Type (i) = 0xdc0000,
+//#   Stateless Reset Tokens [(128)],
+//# }
+
+//# DC_STATELESS_RESET_TOKENS frames contain the following field:
+//#
+//# Stateless Reset Tokens: 1 or more 128-bit values that will be used
+//#     for a stateless reset of dc path secrets
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct DcStatelessResetTokens<'a> {
+    /// 1 or more 128-bit values
+    stateless_reset_tokens: &'a [u8],
```

Comment:
I'd prefer this to be a slice of [`stateless_reset::Token`](https://docs.rs/s2n-quic-core/0.37.0/src/s2n_quic_core/stateless_reset/token.rs.html#26). This should be doable if you derive zerocopy traits for that type.

Reply:
this is going from `&'a [u8]` to `&'a [[u8; 16]]` to `&'a [stateless_reset::Token]`. Would the zerocopy traits have to be derived also for `[stateless_reset::Token]`?

Reply:
https://docs.rs/zerocopy/latest/zerocopy/trait.FromBytes.html#method.slice_from

Reply:
Actually you probably need something like https://docs.rs/zerocopy/latest/zerocopy/trait.FromBytes.html#method.slice_from_prefix

Reply:
Thanks! Happy to bring back the `#![forbid(unsafe_code)]`. It ended up getting a little funky with having a `impl_decode_parameterized` macro, but overall nicer